components/org.wso2.carbon.appmgt.mdm.osgiconnector/src/main/java/org/wso2/carbon/appmgt/mdm/osgiconnector/ApplicationOperationsImpl.java

```diff
@@ -0,0 +1,251 @@
+								.createInstallAppOperation(mobileApp);
+					} else {
+						operation = AndroidApplicationOperationUtil
+								.createAppUninstallOperation(mobileApp);
+					}
+				} else if (deviceIdentifier.getType().equals(Platform.ios.toString())) {
+					if (MDMAppConstants.INSTALL.equals(applicationOperationAction.getAction())) {
+						operation =
+								IOSApplicationOperationUtil.createInstallAppOperation(mobileApp);
+					} else {
+						operation =
+								IOSApplicationOperationUtil.createAppUninstallOperation(mobileApp);
+					}
+				}
+				MDMServiceAPIUtils.getAppManagementService(applicationOperationAction.getTenantId())
+				                  .installApplicationForDevices(operation, deviceIdentifiers);
+			}
+		} catch (DeviceApplicationException mdmExce) {
+			log.error("Error in creating operation object using app.", mdmExce);
+			throw new MobileApplicationException(mdmExce);
+		} catch (ApplicationManagementException appMgtExce) {
+			log.error("Error in app installation.", appMgtExce);
+			throw new MobileApplicationException(appMgtExce);
+		}
+
+	}
+
+	/**
+	 * Create a new device identifier from Device object.
+	 * @param device device which is to be retrieved type and id
+	 * @return created device identifier
+	 */
+	private static DeviceIdentifier getDeviceIdentifierByDevice(
+			org.wso2.carbon.device.mgt.common.Device device) {
+		DeviceIdentifier deviceIdentifier = new DeviceIdentifier();
+		deviceIdentifier.setId(device.getDeviceIdentifier());
+		deviceIdentifier.setType(device.getType());
+
+		return deviceIdentifier;
+	}
+
+	/**
+	 * @param applicationOperationDevice holds the information needs to retrieve device list.
+	 * @return List of devices
+	 * @throws MobileApplicationException
+	 */
+	public List<Device> getDevices(ApplicationOperationDevice applicationOperationDevice)
+			throws MobileApplicationException {
+
+		List<Device> devices;
+		try {
+			List<org.wso2.carbon.device.mgt.common.Device> deviceList = MDMServiceAPIUtils
+					.getDeviceManagementService(applicationOperationDevice.getTenantId()).
+							getDevicesOfUser(
+									applicationOperationDevice.getCurrentUser().getUsername());
+			devices = new ArrayList<>(deviceList.size());
+			if(log.isDebugEnabled()){
+				log.debug("device list got from mdm "+ deviceList.toString());
+			}
+			for (org.wso2.carbon.device.mgt.common.Device commondevice : deviceList) {
+				if (MDMAppConstants.ACTIVE
+						.equals(commondevice.getEnrolmentInfo().getStatus().toString().
+								toLowerCase())) {
+					Device device = new Device();
+					org.wso2.carbon.appmgt.mobile.beans.DeviceIdentifier deviceIdentifier =
+							new org.wso2.carbon.appmgt.mobile.beans.DeviceIdentifier();
+					deviceIdentifier.setId(commondevice.getDeviceIdentifier());
+					deviceIdentifier.setType(commondevice.getType());
+					device.setDeviceIdentifier(deviceIdentifier);
+					device.setName(commondevice.getName());
+					device.setModel(commondevice.getName());
```

Comment:
model is not available for now in device mgt side. For the displaying purpose, name is added. once model attribute available in common device in device mgt repo, this should be fixed.